### ISSUE TITLE
Add an example that shows how to rotate an HMI map

### ISSUE THREAD

**Issue by ehsteve** [sunpy/sunpy#3233]
### Description

HMI fits file need to be rotated by default. This is a common user function and an example would be very useful.

**Comment by ayshih** [comment]
> HMI fits file need to be rotated by default.

To be clear, you're just saying that all HMI maps need to be rotated to have solar north pointing up, and that's something that many users will want to do immediately.  You're not actually making the assertion/request that `HMIMap` should rotate internally as a default action irrespective of what a user might want, correct?

**Comment by Cadair** [comment]
Yeah we should provide a gallery example that makes it clear how to do this, but they don't need to "do it by default" as not all operations need you to have a coordinate system aligned with the grid.

**Comment by wtbarnes** [comment]
Could we just address this in the [existing HMI example](https://docs.sunpy.org/en/stable/generated/gallery/acquiring_data/downloading_hmi.html) or do you think this warrants its own entry in the gallery?

**Comment by PaulJWright** [comment]
I think addressing it at the end of the existing HMI example would be an ideal location for it. Also adding a note in the [map rotation](https://docs.sunpy.org/en/stable/generated/gallery/map/map_rotation.html) mentioning that rotating HMI data is addressed elsewhere?

**Comment by mbobra** [comment]
The [existing HMI example](https://docs.sunpy.org/en/stable/generated/gallery/acquiring_data/downloading_hmi.html) has a few issues:
* It queries for line-of-sight magnetic field data at a cadence of 720s, but downloads the 45s data.
* The wavelength is incorrect (says 6174 Angstroms, not the correct value of 6173).

It also looks like rotating the HMI image seems to work, but somehow Map shows a zoomed-in image. Does anyone know why this is? These are the lines I appended to the end of the example: 
```
hmi_rotated = hmi_map.rotate(angle=180.0 * u.deg)
hmi_rotated.plot()
plt.show()
```

**Comment by wtbarnes** [comment]
>     * It queries for line-of-sight magnetic field data at a cadence of 720s, but downloads the 45s data.

This is a bit confusing, but I think what is happening is that it is sampling the `45_s` series at a 720 s cadence. `attrs.Sample` does not change the series, but rather just the cadence at which the series is queried. I suspect the VSO does not even bother with the `720_s` series as it is already included in `45_s` (I could be wrong about this so someone more familiar with the VSO should probably weigh in too).

>     * The wavelength is incorrect (says 6174 Angstroms, not the correct value of 6173).

This is the response from the VSO. I'm not sure we can do anything about that? It looks like it returns a wavelength range, where the lower bound is 6173 and the upper bound is 6174.

> It also looks like rotating the HMI image seems to work, but somehow Map shows a zoomed-in image. Does anyone know why this is? These are the lines I appended to the end of the example:
> 
> ```
> hmi_rotated = hmi_map.rotate(angle=180.0 * u.deg)
> hmi_rotated.plot()
> plt.show()
> ```

I don't think it is zoomed it. It's just that the default interpolation order (4) changes all NaNs (i.e. the off-disk values) to 0 such that the edges of the disk are no longer obvious. If you pass `order=3` to `rotate` then the NaNs will persist. Additionally, you don't have to pass 180 degrees explicitly to `rotate`. Because it knows the value of `CROTA2` from the header, `rotate` will by default diagonalize the rotation matrix.

**Comment by mbobra** [comment]
> This is a bit confusing, but I think what is happening is that it is sampling the `45_s` series at a 720 s cadence. `attrs.Sample` does not change the series, but rather just the cadence at which the series is queried. I suspect the VSO does not even bother with the `720_s` series as it is already included in `45_s` (I could be wrong about this so someone more familiar with the VSO should probably weigh in too).

The 720s data are not interlaced with the 45s data; they are taken with different cameras, with different noise profiles, and stored in different data series (`hmi.M_720s` versus `hmi.M_45s`). The VSO does not seem to distinguish between these two series. [Both VSO entries that contain HMI data and the keyword `LOS_MAGNETIC_FIELD`](https://sdac.virtualsolar.org/cgi/show_details?physobs=LOS_MAGNETIC_FIELD) point to `hmi.M_45s`.

> This is the response from the VSO. I'm not sure we can do anything about that? It looks like it returns a wavelength range, where the lower bound is 6173 and the upper bound is 6174.

Interesting.

> I don't think it is zoomed it. It's just that the default interpolation order (4) changes all NaNs (i.e. the off-disk values) to 0 such that the edges of the disk are no longer obvious. If you pass `order=3` to `rotate` then the NaNs will persist. Additionally, you don't have to pass 180 degrees explicitly to `rotate`. Because it knows the value of `CROTA2` from the header, `rotate` will by default diagonalize the rotation matrix.

Thanks for your help! 😄 I implemented this change and created a PR (#3573).

**Comment by wtbarnes** [comment]
> The 720s data are not interlaced with the 45s data; they are taken with different cameras, with different noise profiles, and stored in different data series (`hmi.M_720s` versus `hmi.M_45s`). The VSO does not seem to distinguish between these two series. [Both VSO entries that contain HMI data and the keyword `LOS_MAGNETIC_FIELD`](https://sdac.virtualsolar.org/cgi/show_details?physobs=LOS_MAGNETIC_FIELD) point to `hmi.M_45s`.

Oh I see. I was not aware of that. I guess the problem is that `attrs.Sample` does not change the series, but that's not unexpected either. If a user was not aware of the different series names in JSOC, they would probably not be bothered by this. However, maybe we should change the sample cadence to something other than 720 s to avoid this confusion?

**Comment by wtbarnes** [comment]
Closed by #3573
